src/Solana/useSolana.tsx

```diff
@@ -0,0 +1,390 @@
+      vals[accountData.mint.toBase58()] = accountData.amount
+    })
+
+    return {
+      hntBalance: vars?.hnt.mint ? vals[vars.hnt.mint] : 0n,
+      iotBalance: vars?.iot.mint ? vals[vars.iot.mint] : 0n,
+      dcBalance: vars?.dc.mint ? vals[vars.dc.mint] : 0n,
+      mobileBalance: vars?.mobile.mint ? vals[vars.mobile.mint] : 0n,
+    }
+  }, [connection, vars, wallet])
+
+  const getSolBalance = useCallback(async () => {
+    if (!wallet) return 0
+    return connection.getBalance(wallet)
+  }, [connection, wallet])
+
+  const submitSolana = useCallback(
+    async ({ txn }: { txn: Buffer }) => {
+      const { txid } = await sendAndConfirmWithRetry(
+        connection,
+        txn,
+        { skipPreflight: true },
+        'confirmed'
+      )
+      return txid
+    },
+    [connection]
+  )
+
+  const submitAllSolana = useCallback(
+    ({ txns }: { txns: Buffer[] }) =>
+      Promise.all(txns.map((txn) => submitSolana({ txn }))),
+    [submitSolana]
+  )
+
+  const getOraclePriceFromSolana = useCallback(
+    async ({
+      tokenType,
+    }: {
+      tokenType: 'HNT'
+    }): Promise<Currency.PriceData | undefined> =>
+      Currency.getOraclePrice({ tokenType, cluster, connection }),
+    [cluster, connection]
+  )
+
+  const createTransferCompressedCollectableTxn = useCallback(
+    async ({
+      collectable,
+      newOwnerHeliumAddress,
+    }: {
+      collectable: Asset
+      newOwnerHeliumAddress: string
+    }): Promise<VersionedTransaction | undefined> => {
+      if (!wallet) return
+      const recipient = heliumAddressToSolPublicKey(newOwnerHeliumAddress)
+      return Hotspot.createTransferCompressedCollectableTxn({
+        collectable,
+        owner: wallet,
+        recipient,
+        connection,
+        url: METAPLEX_URL,
+      })
+    },
+    [connection, wallet]
+  )
+
+  const getHotspots = useCallback(
+    async (
+      opts: Omit<SearchAssetsOpts, 'ownerAddress' | 'creatorAddress'> & {
+        makerName?: string
+      }
+    ) => {
+      if (!wallet) return
+
+      const searchParams = {
+        ownerAddress: wallet.toString(),
+        ...opts,
+      } as SearchAssetsOpts
+
+      if (vars?.hnt.mint) {
+        const hnt = new PublicKey(vars.hnt.mint)
+        const key = entityCreatorKey(daoKey(hnt)[0])[0].toString()
+        searchParams.creatorAddress = key.toString()
+      }
+
+      if (opts.makerName && hemProgram) {
+        const maker = makerKey(opts.makerName)[0]
+        const makerAcc = await hemProgram.account.makerV0.fetch(
+          maker.toString()
+        )
+        searchParams.collection = makerAcc.collection.toString()
+      }
+
+      return searchAssets(METAPLEX_URL, searchParams)
+    },
+    [hemProgram, vars?.hnt.mint, wallet]
+  )
+
+  const estimateMetaTxnFees = useCallback(
+    async (buff: Buffer, { maker }: { maker: PublicKey }) => {
+      if (!wallet || !vars?.dc.mint) return
+
+      const walletDC = await getAssociatedTokenAddress(
```

Comment:
@ChewingGlass Fees appear to be correct when updating for the first time, but they go haywire the second. 

1. When I update meta for mobile it shows the maker is the payer (should be owner)
2. When I update meta for iot none of the accounts are found in the simulated txn.

I'm guessing it has something to do with them not being free the second time or that I have no dc in my account.

Reply:
I could also be that you didn't change location? Updating meta is free if you're only changing gain, height, etc.